First pull request for the purpose of review

moheed-ali opened the pull request:
This pull request provides the first working version of bike-mobility-example with a waypoint mobility model and periodic sender#

## Proposed Changes

  - Creation of new bike-mobility-helper
  - Creation of new bike-application

moheed-ali committed:
Simple First Example of Constant Velocity Mobility Model

moheed-ali committed:
Simple First Example of WayPoint Mobility Model

moheed-ali committed:
Simple Second Example of Constant Velocity Mobility Model

moheed-ali committed:
Test Stations Data Set

moheed-ali committed:
Test Bikes Data Set

moheed-ali committed:
Test File For Experiments

moheed-ali committed:
Updating The API Key

moheed-ali committed:
Now 5 mobile nodes are in operation and file handling is also working

moheed-ali committed:
Another file of data set with multiple stations

moheed-ali committed:
Update README.md

Adding this repo SSH clone address

moheed-ali committed:
Adding Application class function and application helper function along with Waypoint Mobility

moheed-ali committed:
Real Clean Data Set

moheed-ali committed:
Mobility Example with real data set

moheed-ali committed:
Vector Mobility Model with map and application class

moheed-ali committed:
Testing with one node value

moheed-ali committed:
Some Formatting

moheed-ali committed:
Waypoint Mobility

moheed-ali committed:
Unix to int converted

moheed-ali committed:
Adding new API Key

moheed-ali committed:
Now example has haswaypoints and getvelocity functions as well

moheed-ali committed:
Using two nodes to display the behaviour of the Simulator

moheed-ali committed:
Waypoints are added to mobility objects

moheed-ali committed:
Application class print function is upgraded to use with waypoint model

moheed-ali committed:
Testing Application Class with node 1

moheed-ali committed:
Start time map and end time map is added to the code

moheed-ali committed:
Maps are working good now

moheed-ali committed:
Application Class is working with nodes

moheed-ali committed:
Fixing a error in End Time Map

moheed-ali committed:
changing start and end time variable name in structure

moheed-ali committed:
correct implementation of row++

moheed-ali committed:
Simple implementation of pause

moheed-ali committed:
Some Modifications

moheed-ali committed:
Updating the Dir Structure

moheed-ali committed:
Adding Mobility example to ns3 build

moheed-ali committed:
Removing additional cmake file

moheed-ali committed:
Adding bike-app header and cc file in cmake

moheed-ali committed:
Adding test_ex in cmake file

moheed-ali committed:
Now example has application class from header file

moheed-ali committed:
A file for testing purposes

moheed-ali committed:
Custom header file for bike mobility

moheed-ali committed:
Custom application for bike mobility

moheed-ali committed:
moving the application pointer code

moheed-ali committed:
Bike mobility header file for functions

moheed-ali committed:
Bike mobility helper class  file for functions

moheed-ali committed:
Adding new helper and cc class to the list

moheed-ali committed:
Moving includes to header file

moheed-ali committed:
Moving includes to header file

moheed-ali committed:
Updated file with new header file included

moheed-ali committed:
Adding save mobility function to return the waypoint mobility object

moheed-ali committed:
Adding save mobility function to return the waypoint mobility object

moheed-ali committed:
Updated code with save waypoints function

moheed-ali committed:
Updated code with start node time function

moheed-ali committed:
Updated code with start and end  node time function

moheed-ali committed:
Updated code with start/end node node time function

nondetalle committed:
Add minimal example to be expanded

nondetalle committed:
Add minimal example to be expanded

moheed-ali committed:
Merge remote-tracking branch 'origin/bikes'

moheed-ali committed:
Adding this file from orignal repo to test elora

moheed-ali committed:
Changing the print position functionality with send packets

moheed-ali committed:
Changing the print position functionality with send packets

moheed-ali committed:
Class is Implemented

moheed-ali committed:
Class is Implemented

moheed-ali committed:
Testing the new helper class

moheed-ali committed:
Class is implemented and tested

moheed-ali committed:
Class is implemented and tested

moheed-ali committed:
Adding few comments

moheed-ali committed:
WM without Header files

moheed-ali committed:
File is modified to test bike mobility helper

moheed-ali committed:
Some chnages to test the helper class

moheed-ali committed:
Using Bike Helper and application class in the example

moheed-ali committed:
Creating a install function to install mobility and saving waypoints

moheed-ali committed:
Creating a install function to install mobility and saving waypoints

moheed-ali committed:
Removing comments from send packet

moheed-ali committed:
Creating one app per node

moheed-ali committed:
Increasing the height of the node to 1 meter

moheed-ali committed:
Removing the old dataset

moheed-ali committed:
Adding new updated dataset

moheed-ali committed:
Some modifications to validate the new dataset

moheed-ali committed:
Updating the bike structure and adding some comments

moheed-ali committed:
Updating the read Data function accordingly to the new bike structure

moheed-ali committed:
Updating the path of data set to the new data set csv

non-det-alle commented on CMakeLists.txt:
```diff
@@ -9,6 +9,7 @@ set(source_files
     model/app/server/network-controller.cc
     model/app/server/network-controller-components.cc
     model/app/server/adr-component.cc
+    model/app/bike-application.cc #new entry
```
```suggestion
    model/app/bike-application.cc
```

non-det-alle commented on CMakeLists.txt:
```diff
@@ -18,7 +19,7 @@ set(source_files
     model/mac/gateway-lorawan-mac.cc
     model/mac/base-end-device-lorawan-mac.cc
-    model/mac/class-a-end-device-lorawan-mac.cc
+    model/mac/class-a-end-device-lorawan-mac.cc 
```
```suggestion
    model/mac/class-a-end-device-lorawan-mac.cc
```

nondetalle committed:
Removed comments from CMakeLists
+ moved new entries close to similar classes

nondetalle committed:
Remove unused testing examples

nondetalle committed:
Merge branch 'master' into bikes

nondetalle committed:
Merge remote-tracking branch 'upstream/bikes'

nondetalle committed:
First working version
+ remove old datasets
+ create new example directory
+ rename bike-mobility-helper to bike-sharing-mobility-helper
+ general improvements to
    - bike-sharing-mobility-helper
    - bike-application
    - bike-mobility-example

nondetalle committed:
Add BikeApplicationHelper + comments & formatting

nondetalle committed:
Complete example
+ ordered map for consistentency when loading trips

nondetalle committed:
Merge remote-tracking branch 'upstream/bikes'

nondetalle committed:
Merge remote-tracking branch 'upstream/bikes'

nondetalle committed:
Remove old example

nondetalle committed:
New path loss model + restore api key

nondetalle committed:
Add netanim example
+ dataset file check

non-det-alle commented:
Hello,
I reviewed the code, made some modifications, and added a couple of examples.
I will now merge the branch.
Best,
Alessandro
